### PR TITLE
refactor some calculation from saturating to checked

### PR DESCRIPTION
**Pull request by wangjj9219**
related to #1192 

If DEX opens trading pair registration in the future, especially for these ERC20 tokens, many saturation calculations which use `FixedU128` as multiper are actually easy to overflow, which will bring malicious attacks. So it's necessary to modify those saturation calculations to checked calculations.

### CODE REVIEW

File: modules/cdp-treasury/src/lib.rs [hunk 1]
```diff
@@ -285,7 +285,7 @@ impl<T: Config> CDPTreasury<T::AccountId> for Pallet<T> {
 
 	fn get_debit_proportion(amount: Self::Balance) -> Ratio {
 		let stable_total_supply = T::Currency::total_issuance(T::GetStableCurrencyId::get());
-		Ratio::checked_from_rational(amount, stable_total_supply).unwrap_or_default()
+		Ratio::checked_from_rational(amount, stable_total_supply).unwrap_or_else(Ratio::max_value)
```
Comment (xlc) [inline modules/cdp-treasury/src/lib.rs:288]: return zero makes more sense here

File: modules/staking-pool/src/lib.rs [hunk 2]
```diff
@@ -113,13 +113,15 @@ impl Ledger {
 
 	/// The ratio of `free_pool` in `total_belong_to_liquid_holders`.
 	fn free_pool_ratio(&self) -> Ratio {
-		Ratio::checked_from_rational(self.free_pool, self.total_belong_to_liquid_holders()).unwrap_or_default()
+		Ratio::checked_from_rational(self.free_pool, self.total_belong_to_liquid_holders())
```
Comment (xlc) [inline modules/staking-pool/src/lib.rs:116]: I think it is ok to return zero here?

File: modules/staking-pool/src/lib.rs [hunk 3]
```diff
@@ -113,13 +113,15 @@ impl Ledger {
...
+			.unwrap_or_else(Ratio::max_value)
 	}
 
 	/// The ratio of `unbonding_to_free` in
 	/// `total_belong_to_liquid_holders`.
 	fn unbonding_to_free_ratio(&self) -> Ratio {
-		Ratio::checked_from_rational(self.unbonding_to_free, self.total_belong_to_liquid_holders()).unwrap_or_default()
+		Ratio::checked_from_rational(self.unbonding_to_free, self.total_belong_to_liquid_holders())
```
Comment (xlc) [inline modules/staking-pool/src/lib.rs:123]: I think it is ok to return zero here?

File: modules/cdp-treasury/src/lib.rs (hunk 1 above)
Comment (wangjj9219) [reply modules/cdp-treasury/src/lib.rs:288]: yup, i think too much

File: modules/dex/src/lib.rs [hunk 4]
```diff
@@ -835,6 +852,11 @@ impl<T: Config> Pallet<T> {
 			Error::<T>::MustBeEnabled,
 		);
 
+		ensure!(
+			!max_amount_a.is_zero() && !max_amount_b.is_zero(),
+			Error::<T>::InvalidLiquidityIncrement
+		);
```
Comment (shaunxw) [inline modules/dex/src/lib.rs:858]: Do we need new unit tests to catch this error?

**Review by shaunxw — APPROVED** [review]
LGTM

**Pull request merged**